plugins/woocommerce/tests/php/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsServiceTest.php

```diff
@@ -0,0 +1,4903 @@
+
+		// Assert.
+		$this->assertIsArray( $result );
+		$this->assertArrayHasKey( 'steps', $result );
+		$this->assertCount( count( $expected_steps ), $result['steps'] );
+		$this->assertEquals( $expected_steps, $result['steps'] );
+	}
+
+	/**
+	 * Data provider for test_get_onboarding_details_steps.
+	 *
+	 * @return array[]
+	 */
+	public function provider_get_onboarding_details_steps(): array {
+		$default_recommended_pms = array(
+			array(
+				'id'       => 'card',
+				'enabled'  => false,
+				'required' => true,
+			),
+			array(
+				'id'       => 'apple_pay',
+				'enabled'  => true,
+				'required' => false,
+			),
+			array(
+				'id'       => 'google_pay',
+				'enabled'  => false,
+				'required' => false,
+			),
+		);
+		$expected_pms_state      = array(
+			'card'       => true, // Force enabled because it is required.
+			'apple_pay'  => true,
+			'google_pay' => false,
+		);
+
+		$default_wpcom_connection        = array(
+			'is_store_connected'  => false,
+			'has_connected_owner' => false,
+			'is_connection_owner' => false,
+		);
+		$expected_wpcom_connection_state = array(
+			'has_working_connection' => false,
+			'is_store_connected'     => false,
+			'has_connected_owner'    => false,
+			'is_connection_owner'    => false,
+		);
+
+		$default_account_state = array(
+			'has_account'       => false,
+			'has_valid_account' => false,
+			'test_account'      => false,
+		);
+
+		return array(
+			'clean slate'             => array(
+				array(
+					WooPaymentsService::ONBOARDING_STEP_PAYMENT_METHODS => WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+					WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION   => WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+					WooPaymentsService::ONBOARDING_STEP_TEST_ACCOUNT       => WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+					WooPaymentsService::ONBOARDING_STEP_BUSINESS_VERIFICATION => WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+				),
+				// No profile steps stored details.
+				array(),
+				$default_recommended_pms,
+				$expected_pms_state,
+				$default_wpcom_connection,
+				$expected_wpcom_connection_state,
+				$default_account_state,
+			),
+			'stored statuses (started) - no WPCOM connection, no account' => array(
+				array(
+					WooPaymentsService::ONBOARDING_STEP_PAYMENT_METHODS => WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED,
+					WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION   => WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED,
+					WooPaymentsService::ONBOARDING_STEP_TEST_ACCOUNT       => WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED,
+					WooPaymentsService::ONBOARDING_STEP_BUSINESS_VERIFICATION => WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED,
+				),
+				array(
+					WooPaymentsService::ONBOARDING_STEP_PAYMENT_METHODS => array(
+						'statuses' => array(
+							WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time(),
```

Comment:
maybe mock time so it does not get flaky

Reply:
True. Will wrap the WooPayments `time()` calls in `LegacyProxy::call_function()` and use a test property.